static/backbone/js/api/lib/maps/data/viewLoader.js

```diff
@@ -0,0 +1,131 @@
+/**
+ * Created by zmmachar on 12/17/14.
+ */
+define(["models/view",
+        "collections/projects",
+        "jquery",
+        "config"
+    ],
+    function (View, Projects, $, Config) {
+        'use strict';
+
+        /**
+         * The Viewloader class just structes the data
+         * necessary to load a view
+         * @class ViewLoader
+         */
+        var ViewLoader = function (app) {
+            var updateCollection = function (opts) {
+                if (!opts) {
+                    //TODO: create standardized way to report errors.
+                    return;
+                }
+                var key = opts.key,
+                    models = opts.models,
+                    collectionOpts;
+                if (!this.collections[key]) {
+                    collectionOpts = { key: key, name: opts.name };
+                    //A few special hacks for form data:
+                    if (key.indexOf("form") !== -1) {
+                        collectionOpts.url = '/api/0/forms/' + key.split("_")[1] + '/data/';
+                    }
+                    this.collections[key] = new opts.Collection([], collectionOpts);
+
+                    // important: this trigger enables the overlayManager
+                    // to create a new overlay for each model where the
+                    // GeoJSON geometry is defined.
+                    this.app.vent.trigger('new-collection-created', {collection: this.collections[key],
+                                                                     startVisible: true});
+
+                }
+                this.collections[key].add(models, {merge: true});
+            };
+            /**
+             * A dictionary of the various data types available (given
+             * the projects that have been selected), and the corresponding
+             * data records that have been pulled down from the Data API.
+             */
+            this.collections = {};
+
+            this.initialize = function (opts) {
+                this.view = new View(opts.view);
+                this.app = opts.app;
+                var that = this;
+                this.app.vent.once('map-ready', function () {
```

Comment:
A quick aside, though:  Unit tests are not super appropriate for this sort of application wide manipulation - in particular the expected behavior here is a manipulation of the map object, which is a huge hassle to test headlessly anyway - google maps depends on DOM manipulation and you are going to have a hell of a time writing many tests at all for the basemap that actually involve the map (we can check to see if things have been called in basemap, but it's tough to respond to map-related events like this)